client/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list.jsx

```diff
@@ -0,0 +1,173 @@
+import { Button, Gridicon } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+import PropTypes from 'prop-types';
+import { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { validateAllFields } from 'calypso/lib/domains/email-forwarding';
+import EmailForwardingAddNewCompact from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact';
+import { emailManagement } from 'calypso/my-sites/email/paths';
+import {
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'calypso/state/analytics/actions';
+import { addEmailForward } from 'calypso/state/email-forwarding/actions';
+import getEmailForwardingLimit from 'calypso/state/selectors/get-email-forwarding-limit';
+import getEmailForwardingType from 'calypso/state/selectors/get-email-forwarding-type';
+import { getEmailForwards, isAddingForward } from 'calypso/state/selectors/get-email-forwards';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+class EmailForwardingAddNewCompactList extends Component {
+	static propTypes = {
+		emailForwards: PropTypes.array,
+		emailForwardingLimit: PropTypes.number,
+		selectedDomainName: PropTypes.string.isRequired,
+	};
+
+	state = {
+		forwards: [ { destination: '', mailbox: '', valid: false } ],
+	};
+
+	hasForwards() {
+		return this.props.emailForwards.length > 0;
+	}
+
+	hasReachedLimit() {
+		return this.props.emailForwards.length >= this.props.emailForwardingLimit;
+	}
+
+	validForwards() {
+		return ! this.state.forwards.some( ( t ) => ! t.valid );
```

Comment:
We usually use a full word for variable names:

```suggestion
		return ! this.state.forwards.some( ( forward ) => ! forward.valid );
```

Reply:
Ups I'm using to this naming in functional programming